File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.cpp
class Solution {
public:
    int partitionString(string s) {
        int n = s.size();
        map<char, bool> maps;
        int ans = 1;
        for (int i = 0; i < n; i++) {
            if (maps.count(s[i])) {
                ans++;
                maps.clear();
            }
            maps[s[i]] = true;
        }
        return ans;
    }
};

File: 2405-optimal-partition-of-string/2405-optimal-partition-of-string.py
class Solution:
    def partitionString(self, s: str) -> int:
        seen = [-1] * 26
        res, last_starting = 1, 0
        for i, c in enumerate(s):
            if seen[ord(c) - ord('a')] >= last_starting:
                print(i)
                res += 1
                last_starting = i
            seen[ord(c) - ord('a')] = i
        return res